src/data/supervisor/supervisor.ts

```diff
@@ -70,6 +70,18 @@ export interface Supervisor {
   localize: LocalizeFunc;
 }
 
+export interface SupervisorAvailableUpdates {
```

Comment:
Instead of defining 1 interface, define one for each type and create a union type. That way TypeScript can give better suggestions once you've checked `update_type`.